controller/daemon.go

```diff
@@ -95,8 +136,6 @@ func (ctrl *Daemon) Run(ctx context.Context) error {
 		counter++
 		if err != nil && err != context.Canceled {
 			// Exit if an error is returned
-			// Not expecting any routines to send a nil error because they run
-			// until canceled. Nil check is just to be safe
```

Comment:
Yeah, because we don't want CTS to exit if registration errors, I'm returning always sending `nil` on to the exit channel when the self-manager goroutine ends. This would either be on a graceful exit where no error is returned by Start() or if registration or deregistration fails in which case an error is returned by Start() but we're ignoring it.

https://github.com/hashicorp/consul-terraform-sync/pull/832/files#diff-557dd1981617dc7e0ec1c5d81fda203c5dc5325fa3c59b5ff23083c8b39bdd95R106

Reply:
Ah, thank you for explaining and pointing out the nil to exitCh. That makes sense